### PR TITLE
SWIK-1687_new_route_for_profilepicture

### PR DESCRIPTION
**Pull request by TBoonX**

### CODE REVIEW

**Comment by TBoonX** [comment]
Ready to be merged?

File: application/controllers/picture.js [hunk 1]
```diff
@@ -33,6 +33,17 @@ module.exports = {
       request.log(err);
       return new Promise((resolve, reject) => resolve(boom.badImplementation()));
     }
+  },
+
+  saveProfilepicture: function(request) {
+    let userName = request.auth.credentials.username.toLowerCase();
+    let buffer = readChunk.sync(request.payload.path, 0, 262);
+    if(!['image/jpeg', 'image/png', 'image/tiff', 'image/bmp'].includes(!co.isEmpty(fileType(buffer)) ? fileType(buffer).mime : null))
```
Comment (rmeissn) [inline application/controllers/picture.js]: Please restrict this to png only

**Review by rmeissn — CHANGES_REQUESTED** [review]
Looks good so far.

**Comment by rmeissn** [comment]
I've tested the software:
* If I issue a file and set as content-type header e.g. "image/jpeg", the image is stored to the /tmp directory but not deleted as the request is aborted - this might be an error in other routes too.
* There is no size restriction for the image, except for the file-size (if I upload Full-HD, it's stored Full-HD) --> restrict to some max size, otherwise abort
* There is no aspect ratio restriction for the images --> restrict to 1:1, otherwise abort
* Response is only a string, containing the relative URL --> It might be better to return json, as we also do so in other routes. Maybe discuss about it.

**Comment by TBoonX** [comment]
@rmeissn 
* I added that the file is deleted in every case in the handler.js, but if there is an error deeper inside hapi then the file is not deleted. Thats a general issue for whom I didnt found a solution yet.
* I will not restrict the width or height because the image cropper in the platforms also creates different values. Restrict the file size is enough.
* I added a ratio checking
* I made it a JSON, its better for later enhancements

**Comment by TBoonX** [comment]
@rmeissn do you approve the merge?

**Comment by rmeissn** [comment]
According to https://github.com/slidewiki/slidewiki-platform/blob/35a9ee251eeba8ee9d1cec87c4e8fc5d55989aee/components/User/UserProfile/ChangePictureModal.js#L49 , images will not be any bigger than 170x170. I think it's thus safe to restrict images to max. 300x300px.

**Comment by TBoonX** [comment]
@rmeissn maxWidth is no official attribute of the cropper lib. I tested the cropper with big pictures (filesize and dimensions) and the cropper created different dimension out of it. e.g. 508x508

**Review by rmeissn — APPROVED** [review]

**Pull request merged**